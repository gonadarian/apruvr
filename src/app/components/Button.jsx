/* @flow */
import React, { type Element } from 'react';
import { iif } from '../utils';

type PropsType = {
    name: string,
    isActive?: boolean,
    onClick: () => void,
};

export const Button = ({ name, isActive = false, onClick }: PropsType): Element<*> =>
    <div
        className={iif(isActive, 'btn btn-primary', 'btn btn-default')}
        onClick={onClick}>
        {name}
    </div>;

export default Button;
