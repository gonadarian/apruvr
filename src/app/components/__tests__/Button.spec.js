import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Button component', () => {

    it('renders correctly by default', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.type).toEqual('div');
        expect(tree.props.className).toEqual(expect.stringContaining('default'));
    });

    it('renders correctly when inactive', () => {
        const tree = renderer.create(<Button isActive={false} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.className).toEqual(expect.stringContaining('default'));
    });

    it('renders correctly when active', () => {
        const tree = renderer.create(<Button isActive />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.className).toEqual(expect.stringContaining('primary'));
    });

});
