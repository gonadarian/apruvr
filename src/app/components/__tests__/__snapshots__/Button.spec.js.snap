// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly by default 1`] = `
<div
  className="btn btn-default"
/>
`;

exports[`Button component renders correctly when active 1`] = `
<div
  className="btn btn-primary"
/>
`;

exports[`Button component renders correctly when inactive 1`] = `
<div
  className="btn btn-default"
/>
`;
